diff --git a/entityreference.module b/entityreference.module
index 5f03c5b..b1c45e5 100644
--- a/entityreference.module
+++ b/entityreference.module
@@ -1173,7 +1173,7 @@ function entityreference_field_formatter_prepare_view($entity_type, $entities, $
 
     foreach ($items[$id] as $delta => $item) {
       // Check whether the referenced entity could be loaded.
-      if (isset($target_entities[$item['target_id']])) {
+      if (isset($item['target_id']) && isset($target_entities[$item['target_id']])) {
         // Replace the instance value with the term data.
         $items[$id][$delta]['entity'] = $target_entities[$item['target_id']];
         // Check whether the user has access to the referenced entity.
